### ISSUE TITLE
Ruff treats `pass` and `...` differently for 

### ISSUE THREAD

**Issue by ssteinerx** [astral-sh/ruff#12799]
*Ruff version*: `ruff 0.5.7`

I searched for "pass", "...", "blank-line-between-methods", and "E301" but found nothing.

The example at [blank-line-between-methods (E301)](https://docs.astral.sh/ruff/rules/blank-line-between-methods/#blank-line-between-methods-e301) shows insertion of blank lines between methods of a class.

With `--preview` and `E301` enabled, missing blank lines are detected and fixed.

If, instead of `pass`, you use `...`, then the missing blank between methods is not detected or corrected.  There is a mention in the doc about the rule not enabled in `.pyi` files; this isn't one of those.

```python
# This will get spaced out as per the example for E301
class MyClass(object):
    def func1():
        pass
    def func2():
        pass
```
Becomes:
```python
class MyClass(object):
    def func1():
        pass

    def func2():
        pass
```
```python
# This will NOT get the missing blank lines detected or inserted
# It doesn't matter whether the `...` is on the same line as the `def` or not
class MyClass(object):
    def func1(): ...
    def func2():
        ...
```

**Comment by MichaReiser** [comment]
Hi

This behavior was added in https://github.com/astral-sh/ruff/pull/10704 to make the rule [compatible with the formatter]( https://github.com/astral-sh/ruff/issues/10211). Mentioning this behavior in the documentation does make sense.

**Comment by ssteinerx** [comment]
I want the code formatted the same as everything else with the dummy body (or the other dummy body, whichever is which) on next line, indented, and a space between it and the next method.  Oddly enough, I would have expected the current behaviour for an overload annotation with `...`; go figure.

Since, from reading the related issues, this looks like a horrendous PITA, and is likely to get even more annoying in the future, just documenting that `...` and `pass` are treated differently should be fine.
